utils/helpers.js

```diff
@@ -28,6 +28,21 @@ const parseProtocol = uri => {
   }
 }
 
+/**
+ * Retrieves a random IPFS gateway to alleviate rate-throttling from using only a single gateway.
+ * @returns A random public IPFS gateway
+ */
+const getIPFSGateway = () => {
+  const gateways = [
+    'https://gateway.pinata.cloud',
+    'https://cloudflare-ipfs.com',
+    'https://ipfs.fleek.co',
+    'https://dweb.link'
+  ];
+  const rand = Math.floor(Math.random() * (gateways.length));
```

Comment:
Not sure if getting the gateways completely randomly makes the most sense here. Can we prioritize the most performant gateways to be used first? Something like 80:20:5:5, or 60:25:10:5?

This looks useful: https://github.com/maxim-saplin/ipfs_gateway_research/blob/main/README.md